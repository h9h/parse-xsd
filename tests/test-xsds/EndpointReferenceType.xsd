<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" targetNamespace="http://schemas.xmlsoap.org/ws/2004/08/addressing" elementFormDefault="qualified" blockDefault="#all">
    <!-- //////////////////// WS-Addressing //////////////////// -->
    <!-- Endpoint reference -->
    <xs:element name="EndpointReference" type="wsa:EndpointReferenceType"/>
    <xs:complexType name="EndpointReferenceType">
        <xs:sequence>
            <xs:element name="Address" type="xs:string"/>
            <xs:element name="ReferenceProperties" type="xs:string" minOccurs="0"/>
            <xs:any namespace="##other" processContents="lax" minOccurs="0" maxOccurs="unbounded">
                <xs:annotation>
                    <xs:documentation>
                        If "Policy" elements from namespace "http://schemas.xmlsoap.org/ws/2002/12/policy#policy" are used, they must appear first (before any extensibility elements).
                    </xs:documentation>
                </xs:annotation>
            </xs:any>
        </xs:sequence>
        <xs:anyAttribute namespace="##other" processContents="lax"/>
    </xs:complexType>
</xs:schema>
