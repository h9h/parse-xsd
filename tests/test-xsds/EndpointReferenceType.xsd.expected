{
  "complexType": {
    "anyAttribute": {
      "namespace": "##other",
      "processContents": "lax"
    },
    "name": "EndpointReferenceType",
    "sequence": {
      "any": {
        "annotation": {
          "documentation": "If \"Policy\" elements from namespace \"http://schemas.xmlsoap.org/ws/2002/12/policy#policy\" are used, they must appear first (before any extensibility elements)."
        },
        "maxOccurs": "unbounded",
        "minOccurs": "0",
        "namespace": "##other",
        "processContents": "lax"
      },
      "element": [
        {
          "name": "Address",
          "type": "xs:string"
        },
        {
          "minOccurs": "0",
          "name": "ReferenceProperties",
          "type": "xs:string"
        }
      ]
    }
  },
  "element": {
    "name": "EndpointReference",
    "type": "wsa:EndpointReferenceType"
  },
  "meta": {
    "blockDefault": "#all",
    "elementFormDefault": "qualified",
    "targetNamespace": "http://schemas.xmlsoap.org/ws/2004/08/addressing",
    "xmlns:wsa": "http://schemas.xmlsoap.org/ws/2004/08/addressing",
    "xmlns:xs": "http://www.w3.org/2001/XMLSchema"
  }
}